.travis.yml

```diff
@@ -34,5 +34,15 @@ script:
 after_success:
     - if [[ $TRAVIS_PYTHON_VERSION != pypy* ]]; then pip install coveralls --use-mirrors ; coveralls ; fi
 
+# publish new version to pypi
+deploy:
+  provider: pypi
+  skip_existing: true
+  on:
+    tags: true
+  user: $PYPI_USER
+  password: $PYPI_PASSWORD
+  distributions: "sdist bdist_wheel"
```

Comment:
As a bonus here, you could add a call to [`check-wheel-contents`](https://github.com/jwodder/check-wheel-contents) on the resulting [wheel artifact](https://docs.travis-ci.com/user/deployment/pypi/#releasing-build-artifacts) - in this case it might look like:

```
check-wheel-contents --toplevel=toolz,tlz dist/toolz-*.whl
```

Reply:
This is harder to to because Travis will build source and wheel distributions and then immediately publish it to PyPI, without any place to run this check. But it may be possible to solve it with manually building source and wheels, checking wheels and publishing them using normal commands.

Reply:
True, I did not consider that. I guess at best it would be a "post-upload" sanity check that you could remedy by fixing and releasing a subsequent patch version that addresses whatever issue was found in the wheel.  The alternative would be to execute a job prior to `deploy` that builds the wheel and runs `check-wheel-contents` on that, then only deploy s if `check-wheel-contents` exits 0.  But I concede that's kind of a hassle to build the wheel twice.

Reply:
I though about adding `check-wheel-contents` to `after_deploy` section. However, "`after_deploy` does not affect the status of the build" so you will have to manually check build output.